src/main/scala/com/amazon/deequ/analyzers/catalyst/AttributeReferenceCreation.java

```diff
@@ -87,6 +99,12 @@ public static AttributeReference createSafe(String name) throws IllegalStateExce
                 return (AttributeReference) apply.invoke(companion, name, dataType, true,
                         emptyMetadata, exprId, none, false);
             } else {
+                // Spark 2.4
+                if (Seq.class.isAssignableFrom(apply.getParameterTypes()[5])) {
```

Comment:
would be better to break this up in to several lines and use temporary variables. It is hard to understand whats happening here otherwise.